text/0009-BundleTracker-integration.md

```diff
@@ -0,0 +1,588 @@
+
+#### Using a BundleTracker
+
+Once a `BundleTracker` is opened, there are API methods to gain information and monitor bundles:
+
+- `GetBundles()` returns the `Bundle` objects currently being tracked.
+- `GetObject(Bundle)` returns the custom object paired to a `Bundle`.
+- `GetTracked()` returns all of the bundles and custom objects as an unordered map.
+- `IsEmpty()` tells the client whether bundles are being tracked.
+- `Size()` returns the number of currently tracked bundles.
+- `GetTrackingCount()` returns a tracking number that ticks up every time the `BundleTracker` changes, allowing for comparisons over time.
+
+#### Customizing a BundleTracker
+
+To implement the `BundleTrackerCustomizer` methods, there are two options that have the same effect:
+
+- Implementing a concrete subclass of `BundleTrackerCustomizer` and passing a `std::shared_ptr` to an instance into the `BundleTracker` constructor.
+- Subclassing `BundleTracker` and overriding the `BundleTrackerCustomizer` methods, then passing in a `nullptr` to the `BundleTracker` constructor.
+
+The `BundleTrackerCustomizer` methods offer the ability to create and manage custom objects to be tracked (the pointer returned from `AddingBundle(const Bundle&, const BundleEvent&)` will be passed into the other methods when called on the same `Bundle`). Additionally, if a `NULL` is returned, the object will not be tracked by the `BundleTracker`.
+
+The three callbacks are called under the following conditions:
+
+`AddingBundle`:
+
+- A `Bundle` enters a state covered by the `stateMask`, and is not currently being tracked.
+
+`RemovedBundle`:
+
+- A `Bundle` that is currently being tracked enters a state not covered by the `stateMask`.
+- A `Bundle` is removed manually using `Remove()`
+- A `Bundle` is currently being tracked when the `Close()` method is called on the `BundleTracker` (or when the `BundleTracker` goes out of scope).
+
+`ModifiedBundle`:
+
+- A `Bundle` enters a state covered by the `stateMask`, and is currently being tracked.
+
+#### Extending a BundleTracker
+
+The [OSGi specification](http://docs.osgi.org/specification/osgi.core/7.0.0/util.tracker.html#d0e52020) permits extension of the `BundleTracker` by using wrapper objects. This behavior is mapped to C++ using a template parameter for the `BundleTracker`. By specifying a wrapper object of type `T` to the `BundleTracker`, the user can make the tracker hold on to custom objects through the tracking callbacks that contain extra data or functionality.
+
+## Detailed Design
+****
+### Overview
+
+The `BundleTracker` is implemented as part of of the core framework for CppMicroservices, leveraging existing constructs to listen for fired `BundleEvent` objects and maintaining the map of tracked objects.
```

Comment:
duplicate "of"

>The `BundleTracker` is implemented as part of of the core framework for CppMicroservices

more simply put; The `BundleTracker` is a part of the CppMicroServices core framework

Reply:
The suggested language is used here.